kview/src/main/java/dev/ikm/komet/kview/mvvm/viewmodel/PatternViewModel.java

```diff
@@ -33,12 +41,95 @@ public class PatternViewModel extends FormViewModel {
     public PatternViewModel() {
         super();
-            addProperty(CONCEPT_TOPIC, (UUID) null)
-                    .addProperty(VIEW_PROPERTIES, (ViewProperties) null)
+            addProperty(VIEW_PROPERTIES, (ViewProperties) null)
+                    .addProperty(PATTERN_TOPIC, (UUID) null)
                     .addProperty(STAMP_VIEW_MODEL, (ViewModel) null)
-                    .addProperty(DEFINITION_VIEW_MODEL, (ViewModel) null);
+                    .addProperty(DEFINITION_VIEW_MODEL, (ViewModel) null)
+                    .addProperty(FQN_DESCRIPTION_NAME, (DescrName) null)
+                    .addProperty(OTHER_NAMES, (List) null)
+                    // PATTERN>DEFINITION Purpose and Meaning
+                    .addProperty(PURPOSE_ENTITY, (EntityFacade) null) // this is/will be the 'purpose' concept entity
+                    .addProperty(MEANING_ENTITY, (EntityFacade) null) // this is/will be the 'meaning' concept entity
+                    .addProperty(PURPOSE_TEXT, "")
+                    .addProperty(MEANING_TEXT, "")
+                    .addProperty(PURPOSE_DATE_STR, "")
+                    .addProperty(MEANING_DATE_STR, "")
+                    // PATTERN>DESCRIPTION FQN and Other Name
+                    .addProperty(FQN_DESCRIPTION_NAME_TEXT, "")
+                    .addProperty(OTHER_NAME_DESCRIPTION_NAME_TEXT, "")
+                    // Ordered collection of Fields
+                    .addProperty(FIELDS_COLLECTION, (List<PatternField>) null)
+            ;
+    }
+
+    public void setPurposeAndMeaningText(PatternDefinition patternDefinition) {
+        setPropertyValue(PURPOSE_ENTITY, patternDefinition.purpose());
+        setPropertyValue(MEANING_ENTITY, patternDefinition.meaning());
+
+        String dateAddedStr = LocalDate.now().format(DateTimeFormatter.ofPattern("MMM d, yyyy")).toString();
+        if (patternDefinition.meaning() != null) {
+            setPropertyValue(MEANING_DATE_STR, "Date Added: " + dateAddedStr);
+        }
+        if (patternDefinition.purpose() != null) {
+            setPropertyValue(PURPOSE_DATE_STR, "Date Added: " + dateAddedStr);
+        }
+
+        EntityFacade purposeFacade = getPropertyValue(PURPOSE_ENTITY);
+        EntityFacade meaningFacade = getPropertyValue(MEANING_ENTITY);
+
+        if (purposeFacade != null) {
+            setPropertyValue(PURPOSE_TEXT, purposeFacade.description());
+        }
+        if (meaningFacade != null) {
+            setPropertyValue(MEANING_TEXT, meaningFacade.description());
+        }
+    }
+
+    public void setDescriptionData(PatternField patternFields) {
+
+    }
+
+    public void setFullyQualifiedName(DescrName descrName) {
+        setPropertyValue(FQN_DESCRIPTION_NAME_TEXT, descrName.getNameText());
+    }
+
+    public void setOtherNameText(DescrName descrName) {
+        setPropertyValue(OTHER_NAME_DESCRIPTION_NAME_TEXT, descrName.getNameText());
+    }
+
+    public void addPatternField(PatternField patternField) {
```

Comment:
Can we remove this method? The view model by default will return an observable list.

two possibilities:
```java

ObservableList<PatternField> list = getPropertyValue(FIELDS_COLLECTION);
List<PatternField> list = getPropertyValue(FIELDS_COLLECTION);
```

Reply:
updated